pyfar/dsp/regularization.py

```diff
@@ -0,0 +1,408 @@
+"""
+Documentation for regularization class.
+"""
+
+import pyfar as pf
+from pyfar.dsp.dsp import _cross_fade
+import numpy as np
+import numbers
+
+class RegularizedSpectrumInversion():
+    r"""Class for frequency-dependent regularized inversion.
+
+    Regularized inversion limits the gain applied by an inverse filter.
+    This can be useful to avoid extreme amplification when inverting a signal
+    of limited bandwidth or containing notches in the spectrum.
+
+    The inverse is computed as [#]_:
+
+    .. math::
+        :label: regularized_inversion
+
+        S^{-1}(f) = \frac{S^*(f)}{S^*(f)S(f) + \beta |\epsilon(f)|^2} D(f)
+
+    with :math:`f` being the frequency, :math:`S(f)` the spectrum of the signal
+    to be inverted, :math:`(\cdot)^*` the complex conjugate,
+    :math:`\epsilon(f)` the regularization, and :math:`\beta` a scalar to
+    control the amount of regularization. :math:`D(f)` denotes an optional
+    target function.
+
+    The compensated system :math:`C = S(f)S^{-1}(f)` approaches the target
+    function in magnitude and phase. In many applications, the target function
+    should contain a delay to make sure that :math:`S^{-1}(f)` is causal. The
+    larger :math:`\beta` and :math:`\epsilon(f)` are, the larger the deviation
+    of :math:`C(f)` from the target :math:`D(f)`.
+
+    The inversion is done in two steps:
+
+    1. Define :math:`S(f)`, :math:`\epsilon(f)`, :math:`\beta`, and
+       :math:`D(f)` using one of the ``from_...()`` methods listed below.
+    2. Compute the inverse :math:`S(f)^{-1}` using :py:func:`~invert`
+
+    The parameters that are defined in the first step are often iteratively
+    adjusted. Examples are given in the documentation of the ``from_...()``
+    methods.
+
```

Comment:
```suggestion
```
isnt it nessesary to mention? It is already clear from above and we dont need to teaser the examples here.